src/dfx/src/commands/ledger/account_id.rs

```diff
@@ -36,6 +40,17 @@ pub async fn exec(env: &dyn Environment, opts: AccountIdOpts) -> DfxResult {
         env.get_selected_identity_principal()
             .context("No identity is selected")?
     };
-    println!("{}", AccountIdentifier::new(principal, opts.subaccount));
+    let subaccount = if let Some(subaccount) = opts.subaccount {
+        if opts.subaccount_from_principal.is_some() {
+            return Err(anyhow!(
+                "You can specify at most one of subaccount and subaccount-from-principal arguments."
+            ));
+        }
```

Comment:
Can be handled by clap, e.g. like this: https://github.com/dfinity/sdk/blob/cb67f24b7acb9aaa4609f38a34527a9436fb6b72/src/dfx/src/commands/deploy.rs#L78

Reply:
Thank you for the suggestion! done